modules/gapi/src/compiler/gcompiler.cpp

```diff
@@ -512,7 +517,11 @@ cv::GCompiled cv::gimpl::GCompiler::compile()
 {
     std::unique_ptr<ade::Graph> pG = generateGraph();
     runPasses(*pG);
-    compileIslands(*pG);
+    if (!m_metas.empty())
```

Comment:
Can compilation go a wrong way now? What if I call a regular `compile()` (like in e.g. v4.2) with an empty metadata vector - what will happen?